packages/contents/src/utils.ts

```diff
@@ -0,0 +1,110 @@
+import { PathExt } from '@jupyterlab/coreutils';
```

Comment:
Moving this to a standalone function, outside of a class, means it becomes rather harder to overload... and the original `switch` implementation wasn't very extensible anyway.

It's worth considering this as a new class, advertised as a full interface, and moving the  `switch` monolith into individual, well-typed, documented methods:

```ts
export class DriveContentsProcessor implments IDriveContentsProcessor {
  contentsManager: Contents.IManager
  async processDriveRequest(request) {
    switch(request.method) {
      // ...
    }
  }
  async onReaddir(request: IReadDirRequest): IReadDirResponse {...}
  // ... 
  async onPut(request: IPutRequest): IPutResponse {...}
}
```

This would then be an argument to `BroadCastChannelWrapper`, or a new plugin, depending on whether it's important something else get _exactly_ this instance.

Reply:
I like the idea of splitting the switch into multiple methods of a class 👍🏽 Thanks for the suggestion

Reply:
Though it does not have to be a class, it does not need to keep a state whatsoever. It could simply be free functions

```typescript
export async function readdir(request: IReadDirRequest): IReadDirResponse {...}
  // ... 
export async function put(request: IPutRequest): IPutResponse {...}
```

But I don't have a strong feeling about this.

Reply:
At least the class could keep the content manager as a property

Reply:
... _and_ be subclassable, vs a pile-of-functions, which puts us right back in the "can't be overloaded without extreme feats of monkeypatching" place.

If nothing else, the class-based approach provides better-discoverable, granular documentation for a logical set of tightly-coupled behaviors.

Reply:
I updated the PR introducing the new class.

It's not a singleton provided by a plugin, we could potentially do that in a separate PR